File: 0004.rb
def palindrome? n
  n.to_s === n.to_s.reverse
end

def solve
  largest_palindrome_product = 0

  for i in 111..999
    for j in 111..999
      product = i * j
      if (product > largest_palindrome_product and palindrome? product)
        largest_palindrome_product = product
      end
    end
  end

  largest_palindrome_product
end

puts solve

File: 0004.js
let isPalindrome = number => {
  if (typeof number !== 'string') {
    number = String(number);
  }

  return (number === [...number].reverse().join(''));
}

let solve = () => {
  let largestPalindromeProduct = 0;

  for (let i = 111; i < 1000; i++) {
    for (let j = 111; j < 1000; j++) {
      let product = i * j;
      if (product > largestPalindromeProduct && isPalindrome(product)) {
        largestPalindromeProduct = product;
      }
    }
  }

  return largestPalindromeProduct;
}

console.log(solve());